File: Java labs/lab3/src/com/iot/model/Tea.java
/**
 * 
 */
package com.iot.model;

import model.enums.DrinkType;
import model.enums.DrinkSortType;

/**
 * @author khadai
 *
 */
public class Tea extends Drink {

	private int cupsAmount = 1;

	public Tea() {
	}

	public Tea(String name, int capacity, int price, DrinkSortType sort, int milkAmount, DrinkType drinkType,
			int cupsAmount) {
		super(name, capacity, price, sort, milkAmount, drinkType);
		this.cupsAmount = cupsAmount;
	}

	public int getCupsAmount() {
		return cupsAmount;
	}

	public void setCupsAmount(int cupsAmount) {
		this.cupsAmount = cupsAmount;
	}

	@Override
	public String toString() {
		String info = "tea: " + getName() + ", ";
		info += getCapacity() + "ml, ";
		info += getPrice() + "$, ";
		info += getCupsAmount() + " cups";
		info += ", coffee: 0";
		if (getMilkAmount() > 0) {
			info += ", milk amount: " + getMilkAmount() + "ml ";
		}
		return info;
	}

}

File: Python labs/lab2/Tea.py
from Drink import Drink
from  DrinkType import DrinkType
from  DrinkSortType import DrinkSortType


class Tea(Drink):
    cups_amount = 0
    coffee_amount = 0

    def __init__(self, m_name="Tea", m_capacity=250, m_price=0, m_sort=DrinkSortType.OTHER,
                 m_drink_type=DrinkType.TEA, m_cups_amount=1):
        super().__init__(m_name, m_capacity, m_price, m_sort, m_drink_type)
        self.cups_amount = m_cups_amount

    def reset_values(self, m_name, m_capacity, m_price, m_milk_amount, m_sort, m_drink_type, m_cups_amount):
        super().reset_values(m_name, m_capacity, m_price, m_milk_amount, m_sort, m_drink_type)
        self.cups_amount = m_cups_amount

    def to_string(self):
        return str(self.sort_of_drink.name.lower())+' tea: ' + self.name + ', ' + str(self.capacity) + 'ml, ' + str(self.price) + '$, ' + str(
            self.cups_amount) + ' cups, coffee: 0'

